product_firmware_version/views/product_view.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0"?>
+<odoo>
+    <!-- Product base form view -->
+    <record id="product_template_only_form_fm_version_view" model="ir.ui.view">
+        <field name="name">product.template.product.form</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <data>
+                <field name="barcode" position="after">
+                    <field name="firmware_version"/>
+                </field>
+            </data>
+        </field>
+    </record>
+    <!-- Product base search view -->
+    <record id="product_template_search_fm_version_view" model="ir.ui.view">
+        <field name="name">product.template.search.view</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_search_view"/>
+        <field name="arch" type="xml">
+            <data>
+                <field name="name" position="attributes">
+                    <attribute name="filter_domain">['|','|','|',('default_code','ilike',self),('name','ilike',self),('barcode','ilike',self), ('firmware_version','ilike',self)]</attribute>
```

Comment:
As soon as another addon will have the same idea about adding a field in this domain, only one of them will work. Wouldn't it be safer to add a new search filter? Or maybe customize the name_search?

Reply:
Agree.

We have two options for these.

1. Adding new firmware version filter. (User need to select particular these filter option)

2. Override `name_search`. (only append domain filter)

What your suggestion ?